src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/scope_map.py

```diff
@@ -0,0 +1,145 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.util import CLIError
+from ._utils import get_resource_group_name_by_registry_name
+
+
+def _parse_actions_from_repositories(allow_or_remove_repository):
+    actions = []
+
+    allow_or_remove_repository.sort()
+    for rule in allow_or_remove_repository:
+        splitted = rule.split(';', 1)
+        if len(splitted) != 2:
+            return False, rule
```

Comment:
A better way is to validate all rules before processing this (add a validator). This helps throwing all error messages in all such specified rules together.

Reply:
I see. Indeed, that's a good idea. Will investigate it